### PR TITLE
feature: find jobs endpoint with filter options

### PR DESCRIPTION
**Pull request by SergKukh**
What was done:
- added find job endpoint with filter options
- updated create job endpoint with adding category and skills
- added created_at column to Job entity
- added findCategoryById method to properties service

image

### CODE REVIEW

File: src/modules/jobs/job.service.ts [hunk 1]
```diff
@@ -33,33 +37,103 @@ export class JobsService {
...
   }
 
-  async addJobInfo(
-    payload: AddJobDescriptionDto,
-    user: UserDto,
-  ): Promise<void> {
+  async createJob(payload: CreateJobDto, user: UserDto): Promise<void> {
     try {
-      const currentUser = await this.userService.findByEmail(user.email);
+      const owner = await this.userService.findByEmail(user.email);
```
Comment (lyudad) [inline src/modules/jobs/job.service.ts]: find other way, you already have `user`, its expensive to make query to get same data

File: src/modules/jobs/job.service.ts [hunk 2]
```diff
@@ -33,33 +37,103 @@ export class JobsService {
...
+            ...(category && {
+              category: await this.propertiesService.findCategoryById(category),
+            }),
+          },
+        ])
+        .execute();
 
-      await this.addJobToUser(currentUser, payload);
+      if (skills) {
+        const { id } = insertResult.identifiers.at(0);
```
Comment (lyudad) [inline src/modules/jobs/job.service.ts]: Снимок экрана 2022-11-20 в 20 05 06
item 4

File: src/modules/jobs/job.service.ts (hunk 1 above)
Comment (SergKukh) [reply src/modules/jobs/job.service.ts]: @lyudad fixed

File: src/modules/jobs/job.service.ts (hunk 2 above)
Comment (SergKukh) [reply src/modules/jobs/job.service.ts]: @lyudad fixed with shift() method

**Review by lyudad — APPROVED** [review]

**Pull request merged**